### DISCUSSION TITLE
Setting Up Project Locally

### DISCUSSION THREAD

**Discussion by soniyadotp** [camicroscope/caMicroscope#712] [General]
I am getting problem in setting up the project locally

**Comment by gbengaoluwadahunsi** [comment]
hello @soniyaprasad77 

Kindly follow the following steps


There are three major repos you are to clone
1.https://github.com/camicroscope,  and fork all three repos there, Distro, viewer and Caracal.

2. Fork it and clone it to your machine

3. Install docker via  https://docs.docker.com/engine/install/, after which you start docker to make sure it's running.

4. navigate to your machine where you saved the repos and for the Distro repo, try and build with  docker-compose -f caMicroscope.yml build and run with  docker-compose -f caMicroscope.yml up,

5.  Once everything is up, go to http://localhost:4010/ to see the landing page.

6GB memory is recommended (in total, 2GB RAM + 4GB swap should work well)

6. The frontend files for the caMicroscope app can be found in the viewer repo, navigate to it and just open the index.html file, it will pop up the landing page in your browser, and you can navigate to other pages on the browser.

7. You can then go back to the files and make your contributions based on the following guidelines

-----------------------------------------------------------------------------------------------------------------------------------------------------

Welcome to caMicroscope! These contribution tasks are designed to smoothly onboard contributors, showcase skills, and propel the caMicroscope project forward. Tasks are neatly categorized and can be submitted as pull requests or issues. We recommend starting with the Introductory "Getting Started" and "Open Source Communication" categories.

Tasks beyond the introductory stage are scoped to be achievable within a few days to a week for contributors less familiar with caMicroscope. Remember, quality over quantity! A well-executed contribution, coupled with thorough documentation, speaks volumes about your open-source engagement.

No need to be an expert! These tasks are perfect for developing new skills and gaining hands-on experience. Most importantly, enjoy the process. If challenges arise or you need assistance, don't hesitate to start a discussion thread or open an issue. Take your time, experiment, and, if needed, ask for help (we recommend spending about an hour trying to troubleshoot on your own first; issues which take longer than that are good to document/discuss anyway, and will hopefully stop people from becoming frustrated) Thank you for working with us, and happy contributing!

Introductory: Getting Started (Not Directly Pull Request Changes)



* Clone caMicroscope and its repositories, especially distro
* Install Docker and Docker compose
* Run some version of caMicroscope locally (recommended develop.yml)
* Add a Slide to your instance of caMicroscope
* Experiment with the Viewer Tools
* Experiment with additional Tools

Introductory: Open Source Communication (Not Directly Pull Request Changes)



* Find or start a relevant discussion for a discussion-like topic
* Submit or comment on a relevant issue or pull request
* Specifically, make an issue or pull request on something which was unclear, with a proposed fix if applicable

Open Source Support



* Reproduce a bug on an open issue and comment about how you’ve done so

UI/UX Design and Programming



* Add a dark mode option for the caMicroscope app pages
* Report on a small part of caMicroscope’s UI at a task level and some ways it may be improved from a design perspective, including a plan to evaluate these proposed changes
* Diagnose and fix the caMicroscope viewer toolbar inconsistently resizing toolbar icons with changing browsers and window sizes
* Identify an app or task which is difficult on a mobile device (or other screen size) and propose a low to medium fidelity design solution

Machine Learning and Computer Vision



* Try another way to implement the “smart pens” segmentation annotation tool.
* Report on a relevant model or computer vision algorithm which could be helpful for caMicroscope’s viewer/annotations
* Add a (documented) segmentation ML model to the tfjs_models repository
* Add a (documented) classification ML model to the tfjs_models repository

Accessibility



* identify one accessibility bug using Web Content Accessibility Guidelines (WCAG) 2.1 standard levels within caMicroscope

Code Performance



* Report on a page/app/function which takes a long time for code execution
* In addition to the report, you can also propose an improvement (simpler is better!)
* Implement caching mechanisms to improve the overall speed and responsiveness of caMicroscope.

Testing



* Write tests for some existing caMicroscope functions
* After writing tests, set up github actions’ test suite to run these tests automatically

Happy contributions

**Reply by harshpreet14** [reply]
hey @gbengaoluwadahunsi I followed the exact steps and this is the error I see - 
<img width="710" alt="image" src="https://github.com/camicroscope/caMicroscope/assets/114172812/afe78110-30d0-4aaf-b5f8-80a0166294e8">

**Comment by barchakuz** [comment]
I appricate you sir, can guide me how to create dockerfile for the distro clon repo, i just started working on this project today, i would appricate you time .
note: i have forked and clone all three repo into single folder in my local disk. Installed docker as well.

**Reply by barchakuz** [reply]
issue fixed, I was confused at step i solve it. here is solution for new commers 
once you move to Distro repo in cmd type this "docker-compose -f caMicroscope.yml up --build
"

**Comment by Lahitan** [comment]
@gbengaoluwadahunsi  @barchakuz 
I keep getting an error message when cloning the main Camicroscope repository, it keep saying fatal error, some file was not closed clean, I have cloned carcal and distro repo.

Please  what can I do.

**Reply by gbengaoluwadahunsi** [reply]
hey @Lahitan 

can you walk me through the steps you went through
like
1.
2.
3.

**Reply by barchakuz** [reply]
@Lahitan  Make sure you have fork the repo first and then git clone <url>. Try redoing the fork repo part.

**Reply by ESE-MONDAY** [reply]
After that do we need to do any installation on the viewer repo

**Reply by gbengaoluwadahunsi** [reply]
@ESE-MONDAY , there is no installation for the viewer repo, you just navigate to the index.html file, click on it, and view your landing page on the browser.

**Reply by ESE-MONDAY** [reply]
Hi @gbengaoluwadahunsi  I noticed some difference on my local build and the main website
<img width="245" alt="Screenshot 2024-03-11 131553" src="https://github.com/camicroscope/caMicroscope/assets/63301826/ba6902d5-e480-4f25-9ddc-fcbcae6d1c66">

This is all the landing page sections i'm getting. could there be something wrong

**Reply by birm** [reply]
@ESE-MONDAY  - that looks about right; we may have cleaned up the landing page from an older screenshot. Is something specific concerning you?

**Reply by ESE-MONDAY** [reply]
I was confused because it didn't look like what was on the the live url and i was also hoping to work on the issue -https://github.com/camicroscope/caMicroscope/issues/700

**Reply by Lahitan** [reply]
@gbengaoluwadahunsi  @barchakuz I have successfully set up my environment.
Thank you for your assistance

**Comment by gbengaoluwadahunsi** [comment]
Hello @Ese,

Do you mean the Distro listening port and the frontend caMicroscope repo?

**Reply by ESE-MONDAY** [reply]
yes i do

**Reply by harshpreet14** [reply]
hey @gbengaoluwadahunsi , I have been trying to run the project on my local system. I have followed the steps, forked, cloned, modified develop.yml in Distro, and Dockerfile in Caracal. However, when I try to run ` docker-compose -f develop.yml build` I get a file download error from OrthancFramework,  `ERROR [dicomsrv 13/15] RUN ./build_orthanc_indexer.sh $indexer` 

<img width="720" alt="image" src="https://github.com/camicroscope/caMicroscope/assets/114172812/57bcdf0d-147d-4f1a-a66b-aabb9f8ea149">

<img width="687" alt="image" src="https://github.com/camicroscope/caMicroscope/assets/114172812/bdd510b0-c566-4afe-855b-3a3715e81e1f">

**Comment by gbengaoluwadahunsi** [comment]
Ooh yes,

I guess there is still separation concern on the repos, the Distro is used
for ML and the caMicroscope is for frontend development.

I believe it will be merge soon or before it goes public.

You can also ask Birm.

**Reply by Dalu46** [reply]
Hi can you please point the link to me to the discussion channel?

**Reply by birm** [reply]
I don't quite understand what you mean here; which things do you think are being merged, @gbengaoluwadahunsi 
I'd be happy to clarify if I can better understand!

**Reply by gbengaoluwadahunsi** [reply]
Hello @birm, apologies for my statement that seems ambiguous, what I infer is that the Disto repo and the caMicroscope are not two separate websites but once with a different focus at the moment, and as such before this project goes public, you @birm will guide us on the project on how to update the frontend app of the Distro repo.

**Update:** You have clarified my ambiguity with this 
"The two ways I've typically done development are either mounting your caMicroscope cloned dir in Distro (see https://github.com/camicroscope/camicroscope?tab=readme-ov-file#fast-local-changes), or by pushing to a fork used within 'ca-back''s container. (first configure fork and branch under back, then after rebuilding back, you can run docker exec -it ca-back bash, then pull)"

I hope I made myself clearer now.

**Comment by pamisijohn** [comment]
> I am getting problem in setting up the project locally

I had been getting same error, I just followed a suggestion I saw below; I deleted my previously forked repo and attempted the process again, this solved the issue for me.
